### PR TITLE
Make mirakle work with gradle 4.2+

### PR DESCRIPTION
**Pull request by dsvoronin**
Two noticeable changes:
1. Gradle private Clock api changed from 4.1 to 4.2, and now some of the needed classes are package private. I ported gradle code that you using in simple `Timer` class (dropped all abstractions and tests though)
2. Kotlin plugin now see @Nullable annotations from `gradle.startParameter.buildFile`, i chose to use `!!` operator.

Some problems while working with project:
1. repository doesn't contain any codestyle rules in `.idea` dir, so imports optimized by default, maybe add one?
2. I can't run tests locally via `./gradlew plugin-test:test` errors all about localhost and 22 rsync error. Can you write some docs section about what i need to setup before run tests?

### CODE REVIEW

**Comment by skrugly** [comment]
Since `Clock` is private now, let's just make `val startTime = System.currentTimeMillis()` and have top-level function `fun prettyTime(timeInMs: Long): String` from `Clock.java`. It's super easy and we don't need more. 

Will add contributing doc asap.

Thank you for the contribution!

File: plugin/src/main/kotlin/Timer.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Timer {
+
+    private val MILLIS_PER_SECOND = 1000
+    private val MILLIS_PER_MINUTE = 60 * MILLIS_PER_SECOND
+    private val MILLIS_PER_HOUR = 60 * MILLIS_PER_MINUTE
+
+    private val startTime: Long = getCurrentTime()
+
+    val elapsed: String = formatDurationVerbose(getCurrentTime() - startTime)
```
Comment (skrugly) [inline plugin/src/main/kotlin/Timer.kt]: There is a bug btw, property should define `get() = ` 
```
Total time : 0.0 secs
Task uploadToRemote took: 0.0 secs
Task executeOnRemote took: 0.0 secs
Task downloadFromRemote took: 0.0 secs
```

**Comment by skrugly** [comment]
@dsvoronin Contributing doc is [here](https://github.com/Instamotor-Labs/mirakle/blob/development/docs/CONTRIBUTING.md)

**Comment by dsvoronin** [comment]
Thanks! with contributing doc i was able to run tests locally

**Comment by skrugly** [comment]
Nice, thanks.

**Comment by skrugly** [comment]
Will upload new version soon, use one from local maven for now.

**Pull request merged**